xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/liveDataSource.js

```diff
@@ -93,6 +110,10 @@ define('xwiki-livedata-source', ['module', 'jquery'], function(module, $) {
   };
 
   var toLiveData = function(data) {
+    // We reset the request object to null for two reasons:
+    // - avoid keeping an object we don't need anymore in memory, preventing it from being GC'd
+    // - make sure we don't attempt to abort a request that already terminated.
+    entriesRequest = null;
```

Comment:
This is not the right place to reset the request. ``toLiveData`` should not be aware that there is a request. Its sole responsibility is to adapt the given JSON to the JSON expected by Live Data. The request should be reset in ``getEntries`` which should be the only function handling the request.

Moreover, ``toLiveData`` is called only on a successful request. It probably makes sense to reset even the failed requests, which is another argument for moving this to ``getEntries`` in a request "complete" callback.

Finally, note that promise / deferred callbacks are **always async**. Try this in the JavaScript console for instance:

```
jQuery.Deferred().resolve().then(function() {console.log('after')});console.log('before');
```

This means that when ``toLiveData`` is called you could have another (different) request already sent, and thus you would nullify this new request. Even if you move this to ``getEntries`` you should make sure that you reset the request only if it matches the request on which the callback was added. This can be done for instance by keeping a request count (ID) and resetting the request only if it's the last one (its ID equals the request count).

Reply:
> I figured I could bind a clean up function to the current request object, and if it is equal to the stored request by ref, there's no in flight request coming and we can nullify it.

Clever! I like it.